### ISSUE TITLE
ignore_index defined but not used

### ISSUE THREAD

**Issue by adrian-spataru** [lucidrains/DALLE-pytorch#57]
`ignore_index`is defined but not used.
I assume the purpose is to ignore the padding from the text. If that's the case, then instead of using:
`F.cross_entropy(rearrange(logits, 'b n c -> b c n'), labels)`
We have to define a nn.Module Cross_entropy
`self.loss = nn.CrossEntropyLoss(reduction='none', ignore_index=self.ignore_index)`
and then
`self.loss(rearrange(logits, 'b n c -> b c n'), labels)`

Finally the default value of ignore_index should be 0. Since thats the pad_id of the default tokenizer.

**Comment by lucidrains** [comment]
@adrian-spataru ohh yes, thanks for that, i've addressed it in https://github.com/lucidrains/DALLE-pytorch/pull/58/files

**Comment by adrian-spataru** [comment]
I just realized. We would ignore also the VQVAE codebook with id of 0.
Should we just change the padding in the tokenizer to a negative value?

**Comment by lucidrains** [comment]
@adrian-spataru ohh it wouldn't, because i actually bump the range of the image token indices by the text sequence length https://github.com/lucidrains/DALLE-pytorch/blob/main/dalle_pytorch/dalle_pytorch.py#L454

so the logits would be in the order of `[text token ids] ... [image token ids]`

**Comment by afiaka87** [comment]
Can probably close this one, right?
